src/NServiceBus.Testing.Tests/HandlerTests.cs

```diff
@@ -770,6 +855,22 @@ class AsyncHandler : IHandleMessages<MyCommand>
             }
         }
 
+        class ConcurrentHandler : IHandleMessages<MyCommand>
+        {
+            public int NumberOfThreads { get; set; } = 100;
+
+            public Func<IMessageHandlerContext, Task> HandlerAction = x => Task.FromResult(0);
+
+            public Task Handle(MyCommand message, IMessageHandlerContext context)
+            {
+                var operations = new ConcurrentBag<Task>();
+
+                Parallel.For(0, NumberOfThreads, x => operations.Add(HandlerAction(context)));
```

Comment:
I needed a way to ensure we're running concurrently since the whole outgoing pipeline will run synchronously because we're not using any non-cpu bound async operations.